### ISSUE TITLE
Job failed with no valuable logs

### ISSUE THREAD

**Issue by d-walsh** [All-Hands-AI/openhands-resolver#233]
WIth the latest changes my job failed but it doesn't print any logs like it used to:
```
06:37:24 - openhands:INFO: resolve_issue.py:313 - Using output directory: output
Cloning into 'output/repo'...
06:37:26 - openhands:INFO: resolve_issue.py:337 - Base commit: 449affafdcf73d05b33c889a987d22feefcebb80
06:37:26 - openhands:INFO: resolve_issue.py:348 - Writing output to output/output.jsonl
06:37:26 - openhands:INFO: resolve_issue.py:361 - Resolving issue 594 with Agent CodeActAgent, model claude-3-5-sonnet-20241022, max iterations 50.
06:37:26 - openhands:INFO: resolve_issue.py:368 - Checking out to PR branch openhands-fix-issue-586 for issue 594
Switched to a new branch 'openhands-fix-issue-586'
06:37:26 - openhands:INFO: utils.py:125 - Starting evaluation for instance 594.
Hint: run "tail -f output/infer_logs/instance_594.log" to see live logs in a separate shell
ERROR:root:  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/openhands_resolver/resolve_issue.py", line 548, in <module>
    main()
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/openhands_resolver/resolve_issue.py", line 529, in main
    asyncio.run(
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/asyncio/runners.py", line 194, in run
    return runner.run(main)
           ^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/asyncio/base_events.py", line 687, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/openhands_resolver/resolve_issue.py", line 383, in resolve_issue
    output = await process_issue(
             ^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/openhands_resolver/resolve_issue.py", line [23](***/actions/runs/11640631617/job/32418206224#step:9:24)1, in process_issue
    success, comment_success, success_explanation = issue_handler.guess_success(issue, state.history, llm_config)
                                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/hostedtoolcache/Python/3.12.7/x64/lib/python3.12/site-packages/openhands_resolver/issue_definitions.py", line 422, in guess_success
    raise ValueError("Expected review comments to be initialized.")

ERROR:root:<class 'ValueError'>: Expected review comments to be initialized.
Error: Process completed with exit code 1.
```

**Comment by neubig** [comment]
I'm also experiencing this...

I heard from @xingyaoww that this is due to this change to openhands proper: https://github.com/All-Hands-AI/OpenHands/pull/4563

I tried to fix this by adding `DEBUG=True`, but this doesn't seem to be working: https://github.com/All-Hands-AI/openhands-resolver/pull/228

According to @rbren , we may have to explicitly subscribe to the event stream and decide what to print, like:

```python
def on_event(evt):
  print(evt)

event_stream.subscribe(on_event, EventStreamSubscriber.MAIN)
```

Here's a reference: https://github.com/All-Hands-AI/OpenHands/blob/b4afd9f17061cd2fa90563bb46ee5df74667118a/openhands/core/main.py#L190

**Comment by neubig** [comment]
I found the issue, I'll take this.

**Comment by neubig** [comment]
OK @d-walsh , this should be fixed in the just-released v0.2.5 if you're using it through the github action.
If you're running locally, you will need to set the environment variable `DEBUG=True`.
